godotopenxrmeta/src/main/cpp/include/openxr_fb_scene_extension_wrapper.h

```diff
@@ -0,0 +1,119 @@
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include <godot_cpp/classes/open_xr_extension_wrapper_extension.hpp>
+#include <openxr/openxr.h>
+#include <openxr/fb_scene.h>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+#include "util.h"
+
+#include <optional>
+#include <map>
+
+using namespace godot;
+
+struct XrSceneObjectInternal {
+	XrUuidEXT uuid;
+	XrSpace space;
+	std::optional<std::string> label;
+
+	// Vertices and lines on a plane, probably use this for a floor / ceiling as they are irregularly shaped
+	// We store a std::vector instead of XrBoundary2DFB to own the vertex memory
+	std::optional<std::vector<XrVector2f>> boundary2D;
+	// A rectangle containing the whole thing, perfect for desks / tables / play surfaces
+	std::optional<XrRect2Df> boundingBox2D;
+	// 3D box for the whole thing, better for obstacles and other objects not used as a surface
+	std::optional<XrRect3DfFB> boundingBox3D;
+};
+
+// Wrapper for the set of Facebook XR scene extension.
+class OpenXRFbSceneExtensionWrapper : public OpenXRExtensionWrapperExtension {
+	GDCLASS(OpenXRFbSceneExtensionWrapper, OpenXRExtensionWrapperExtension);
+
+public:
+	godot::Dictionary _get_requested_extensions() override;
+
+	void _on_instance_created(uint64_t instance) override;
+
+	void _on_instance_destroyed() override;
+
+	bool is_scene_supported() {
+		return fb_scene_ext;
+	}
+
+	static OpenXRFbSceneExtensionWrapper *get_singleton();
+
+	OpenXRFbSceneExtensionWrapper();
+	~OpenXRFbSceneExtensionWrapper();
+
+	std::optional<std::string> get_semantic_labels(const XrSpace space);
+	void get_shapes(const XrSpace space, XrSceneObjectInternal& object);
```

Comment:
Here and throughout the PR, we should use a `const XrSpace &space` argument instead.

Reply:
hm does that actually make a difference? the XrSpace, like many other XR handles, is just a pointer / uint64_t, not a full object. See openxr.h:

```
#if !defined(XR_DEFINE_HANDLE)
#if (XR_PTR_SIZE == 8)
    #define XR_DEFINE_HANDLE(object) typedef struct object##_T* object;
#else
    #define XR_DEFINE_HANDLE(object) typedef uint64_t object;
#endif
#endif

...

XR_DEFINE_HANDLE(XrSpace)
```
 Will update to match the style of the rest of the codebase